### DISCUSSION TITLE
Return of "watchBlocks" doesn't include transactions if via webSocket

### DISCUSSION THREAD

**Discussion by livelybug** [wevm/viem#1111] [Question]
Hello,

I am testing the [watching-blocks](https://github.com/wagmi-dev/viem/tree/e37e6eca4497bced238445a816d8307eb13e9dab/examples/blocks/watching-blocks) example.

After changing the client's transport from http to webSocket, the block returned doesn't include any transaction hashes.
```
...
  withdrawalsRoot: '0x6343d769caab8bec43a2143b1d4c68eae0919c3987f32cee884f45bbd1b76741',
  transactions: undefined,
  totalDifficulty: null
}
``` 

If the client's transport is http, it works fine.

Testing env:
```
node: v20.5.1
viem: 1.9.3
typescript: 5.0.3
```
Not sure if it's a bug or I miss something.

Thank you

**Comment by alexisnsns** [comment]
Hey! Struggling with this issue too; do you have any updates regarding this?

**Reply by codeninja819** [reply]
Same for me.
